build-system/server/app.js

```diff
@@ -1696,36 +1689,29 @@ function decryptDocumentKey(encryptedDocumentKey) {
 }
 
 // serve local vendor config JSON files
-app.use('(/dist)?/rtv/*/v0/analytics-vendors/:vendor.json', (req, res) => {
-  const {vendor} = req.params;
-  const serveMode = SERVE_MODE;
-
-  if (serveMode === 'cdn') {
-    const vendorUrl = `https://cdn.ampproject.org/v0/analytics-vendors/${vendor}.json`;
-    request(vendorUrl, (error, response) => {
-      if (error) {
-        res.status(404);
-        res.end();
-      } else {
-        res.send(response);
-      }
-    });
-    return;
-  }
+app.use(
+  '(/dist)?/rtv/*/v0/analytics-vendors/:vendor.json',
+  async (req, res) => {
+    const {vendor} = req.params;
+    const serveMode = SERVE_MODE;
+
+    const filePath = `https://cdn.ampproject.org/v0/analytics-vendors/${vendor}.json`;
+    if (await passthroughServeModeCdn(res, filePath)) {
+      return;
+    }
 
-  const max = serveMode === 'default' ? '.max' : '';
-  const localVendorConfigPath = `${pc.cwd()}/dist/v0/analytics-vendors/${vendor}${max}.json`;
+    const max = serveMode === 'default' ? '.max' : '';
+    const localPath = `${pc.cwd()}/dist/v0/analytics-vendors/${vendor}${max}.json`;
 
-  fs.promises
-    .readFile(localVendorConfigPath)
-    .then((file) => {
+    try {
+      const file = await fs.promises.readFile(localPath);
```

Comment:
Suggestion: Use `fs-extra` instead of `fs`. All the base methods are auto-promisified, and you can use shorthand like `const file = await fs.readFile(localPath);`

And in this specific instance, since the file being read is a JSON file, it's also an option to use `await fs.readJson(localPath)` to directly get the JSON object.

Reply:
Since there are unrelated `fs.promises` users in this file, I would consider switching to `fs-extra` a separate task.

> it's also an option to use `await fs.readJson(localPath)` to directly get the JSON object.

We don't want an object since we're sending the contents of the file as a text response.